### ISSUE TITLE
Files downloaded with Chrome cannot be used in Android Pie

### ISSUE THREAD

**Issue by lini** [NativeScript/nativescript-imagepicker#223]
### Which platform(s) does your issue occur on?
- Android version 9
- Pixel 2

### Please, tell us how to recreate the issue in as much detail as possible. 
 - Download a file using Chrome.
 - Open Image Picker demo.
 - Go to Downloads, select file.
 - App crashes with the following exception:

```bash
JS: Error: java.lang.IllegalArgumentException: Unknown URI: content://downloads/public_downloads/24
JS:     android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:165)
JS:     android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
JS:     android.content.ContentProviderProxy.query(ContentProviderNative.java:418)
JS:     android.content.ContentResolver.query(ContentResolver.java:802)
JS:     android.content.ContentResolver.query(ContentResolver.java:752)
JS:     android.content.ContentResolver.query(ContentResolver.java:710)
JS:     com.tns.Runtime.callJSMethodNative(Native Method)
JS:     com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1116)
JS:     com.tns.Runtime.callJSMethodImpl(Runtime.java:996)
JS:     com.tns.Runtime.callJSMethod(Runtime.java:983)
JS:     com.tns.Runtime.callJSMethod(Runtime.java:967)
JS:     com.tns.Runtime.callJSMethod(Runtime.java:959)
JS:     com.tns.NativeScriptActivity.onActivityResult(NativeScriptActivity.java:66)JS:     android.app.Activity.dispatchActivityResult(Activity.java:7454)
JS:    ...
```

**Comment by leoncc** [comment]
In imagepicker.android.ts we have:

```
    private static getDataColumn(uri: android.net.Uri, selection, selectionArgs) {
        let cursor = null;
        let columns = [android.provider.MediaStore.MediaColumns.DATA];
        let filePath;

        try {
            cursor = this.getContentResolver().query(uri, columns, selection, selectionArgs, null);
            if (cursor != null && cursor.moveToFirst()) {
                let column_index = cursor.getColumnIndexOrThrow(columns[0]);
                filePath = cursor.getString(column_index);
     ......
     ......
```
On Android 7, the columns for the rows of **cursor** are:
_id, _data, _size, _display_name, mime_type, title, date_added, date_modified.............

On Android 8.1 (I have a Go edition) the _data and a lot of the other columns are not there anymore.

So in the query it is looking for _data, but it does not exist.

I guess the only way to fix this is to copy the contents to a stream and save it in a cache directory or something.

**Comment by stevenspiel** [comment]
I'm also having this issue. What workaround would you suggest for now?

**Comment by leoncc** [comment]
I did a ugly one. I replaced ..../node_modules/nativescript-imagepicker/imagespicker.android.js with the file in the zip below. As you know with the next tns plugin add... it will be gone.


[imagepicker.android.js.zip](https://github.com/NativeScript/nativescript-imagepicker/files/2658500/imagepicker.android.js.zip)

**Comment by stevenspiel** [comment]
Thanks @leoncc! I replaced the contents of the file, but I'm still getting the error. Is there some other npm command I need to run to register the changes? If not, I'll just hold out for the next release.

**Comment by leoncc** [comment]
Try deleting the platforms folder and then do tns run android again.

**Comment by RemCom** [comment]
I also run into the same issue as described. The temporary fix posted in the file by @stevenspiel is working for me but it would by nice to have a more permanent solution in the repo.

**Comment by lennonphosfato** [comment]
I'm having the same problem the plugin will be updated with this solution above?

**Comment by Jarrich** [comment]
Same issue here!

**Comment by VladimirAmiorkov** [comment]
Hi all,

This issue has not been added to our immediate plans so if anyone is upto the task if contributing to this project we will be more than happy to review and merge a PR that resolves the above issue. For details on how you could contribute read [this section](https://github.com/NativeScript/nativescript-imagepicker/blob/master/CONTRIBUTING.md#submitting-a-pr).

**Comment by zbranzov** [comment]
The fix is published in nativescript-imagepicker@6.3.0
